packages/workspace/test/workspace/path_index.test.ts

```diff
@@ -536,3 +536,227 @@ describe('updatePathIndex', () => {
     ))
   })
 })
+
+describe('getElementsPathHints', () => {
+  const elemId = new ElemID('salto', 'obj')
+  it('should return one path hint for single path ObjectType with no annotations and fields', async () => {
+    const singlePath = new ObjectType({
+      elemID: elemId,
+      path: ['salto', 'obj', 'simple'],
+    })
+    const pathHints = getElementsPathHints([singlePath])
+    expect(pathHints.length).toEqual(1)
+    expect(pathHints).toEqual(expect.arrayContaining([
+      expect.objectContaining({ key: 'salto.obj', value: [['salto', 'obj', 'simple']] }),
+    ]))
+  })
+
+  it('should return one path hint for annotations and one for fields in ObjectType', async () => {
+    const singleFieldObj = new ObjectType({
+      elemID: elemId,
+      fields: {
+        uriField: {
+          refType: createRefToElmWithValue(BuiltinTypes.STRING),
+          annotations: {
+            test: 'test',
+          },
+        },
+      },
+      path: ['salto', 'obj', 'field'],
+    })
+
+    const singleFieldObjAnnotations = new ObjectType({
+      elemID: elemId,
+      annotationRefsOrTypes: {
+        anno: createRefToElmWithValue(BuiltinTypes.STRING),
+      },
+      annotations: {
+        anno: 'Bye',
+      },
+      path: ['salto', 'obj', 'annotations'],
+    })
+    const pathHints = getElementsPathHints([singleFieldObj, singleFieldObjAnnotations])
+    expect(pathHints.length).toEqual(4)
```

Comment:
I think we don't need this check because this should be covered by the "toEqual" check below IIUC, but for future reference, when asserting the length of an array:
```suggestion
    expect(pathHints).toHaveLength(4)
```